calyx-py/test/correctness/piezo_pifotree.py

```diff
@@ -32,24 +37,39 @@ def insert_stats(prog, name):
     count_1_sto = stats.reg("count_1_sto", 32)
 
     # Wiring to increment the appropriate register.
-    count_0_incr = stats.incr(count_0_sto)
-    count_1_incr = stats.incr(count_1_sto)
+    if static:
+        count_0_incr = stats.incr_static(count_0_sto)
+        count_1_incr = stats.incr_static(count_1_sto)
+    else:
+        count_0_incr = stats.incr(count_0_sto)
+        count_1_incr = stats.incr(count_1_sto)
+
+    # If not static, we can use comb groups.
+    if not static:
+        flow_eq_0 = stats.eq_use(flow, 0)
+
+        with stats.continuous:
+            stats.this().count_0 = count_0_sto.out
+            stats.this().count_1 = count_1_sto.out
+
+        stats.control += cb.par(
+            cb.if_with(flow_eq_0, count_0_incr, count_1_incr),
+        )
+
+        return stats
 
-    # Equality checks.
-    flow_eq_0 = stats.eq_use(flow, 0)
-    flow_eq_1 = stats.eq_use(flow, 1)
+    # If static, we need to use continuous assignments.
+    eq_cell = stats.eq(1, "eq_cell")
 
     with stats.continuous:
         stats.this().count_0 = count_0_sto.out
         stats.this().count_1 = count_1_sto.out
+        eq_cell.left = flow
```

Comment:
For now I have just written, desugared, the code that I needed. It is easy to imagine a little sugar like:
`static_if_while(CellAndGroup, IfBody, ElseBody)`, that 
1. snaps apart the CellAndGroup into `Cell` and `Group`.
2. Puts all of `Group`'s assignments into continuous assignment land.
3. Finally turns itself into `static_if(Cell.out, IfBody, ElseBody)`